WalletWasabi/WabiSabi/Client/CoinJoin/Manager/CoinJoinManager.cs

```diff
@@ -539,13 +539,13 @@ private async Task HandleCoinJoinFinalizationAsync(CoinJoinTracker finishedCoinJ
 			{
 				_coinRefrigerator.Freeze(successfulCoinjoin.Coins);
 				batchedPayments.MovePaymentsToFinished(successfulCoinjoin.UnsignedCoinJoin.GetHash());
-				MarkDestinationsUsed(destinationProvider, successfulCoinjoin.OutputScripts);
 				wallet.LogInfo($"{nameof(CoinJoinClient)} finished. Coinjoin transaction was broadcast.");
 			}
 			else
 			{
 				wallet.LogInfo($"{nameof(CoinJoinClient)} finished. Coinjoin transaction was not broadcast.");
 			}
+			MarkDestinationsUsed(destinationProvider, result.OutputScripts);
```

Comment:
Wouldn't this call be better in the `finally` block? 
Or is there exceptions when we don't want to mark them as used?

Reply:
No because many (most?) errors/exceptions happen before registering any output. In fact, `CoinjoinClientExceptions` are thrown to indicate that it is not possible to participate in a round.